packages/components/src/internal/components/forms/input/DatePickerInput.tsx

```diff
@@ -170,10 +170,14 @@ export class DatePickerInputImpl extends DisableableInput<DatePickerInputImplPro
         return parseDate(value, dateFormat, minDate, false, queryColumn.isDateOnlyColumn);
     }
 
-    onChange = (date: Date, event?: any): void => {
+    onChange = (date: Date, event?: any, raw?: boolean): void => {
         const { formsy, hideTime, inlineEdit, queryColumn } = this.props;
         this.setState({ selectedDate: date, invalid: false, invalidStart: false });
 
+        if (queryColumn.isTimeColumn && !event && !raw && date?.getMilliseconds() > 0) {
+            date.setMilliseconds(0); // react-datepicker milliseconds are not 0 when selecting time
```

Comment:
Here we're modifying the date but two lines above it was just put on the `state.selectedDate`. Side-effecting the date value here and then handing off downstream to other callers seems fragile at best.

Should we just always be zeroing out the milliseconds for any value procured from `DatePicker`?

Reply:
change made to also zeroing out datetime milliseconds. Also fixed the setState call with date.